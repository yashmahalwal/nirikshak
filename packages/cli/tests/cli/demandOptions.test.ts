import path from "path";
import fs from "fs-extra";
import childProcess from "child_process";

const modulePath = path.resolve(__dirname, "../../lib/index.js");

beforeAll(() => {
    if (!fs.existsSync(modulePath))
        childProcess.execSync("cd packages/cli && npm run build:dev");
    process.chdir(__dirname);
});

test("Demand atleast one option", async () => {
    const output = childProcess.execSync(`node ${modulePath}`);
    expect(output.toString()).toMatchSnapshot();
});
