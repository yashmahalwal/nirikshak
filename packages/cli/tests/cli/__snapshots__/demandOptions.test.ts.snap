// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demand atleast one option 1`] = `
"[97m  _   _   _          _   _            _               _    [39m
[97m | \\\\ | | (_)  _ __  (_) | | __  ___  | |__     __ _  | | __[39m
[97m |  \\\\| | | | | '__| | | | |/ / / __| | '_ \\\\   / _\` | | |/ /[39m
[97m | |\\\\  | | | | |    | | |   <  \\\\__ \\\\ | | | | | (_| | |   < [39m
[97m |_| \\\\_| |_| |_|    |_| |_|\\\\_\\\\ |___/ |_| |_|  \\\\__,_| |_|\\\\_\\\\[39m
[97m                                                           [39m
[4m[1mTest your REST APIs with ease[22m[24m


[31mâœ– [39m [31m[4mfatal[24m    [39m YError: Error parsing configuration: data should have required property 'app' [90m[39m
[90m    at Object.parseArgs [as _parseArgs] (/home/yash/Desktop/nirikshak/packages/cli/node_modules/yargs/yargs.js:1232:33)[39m
[90m    at Object.parse (/home/yash/Desktop/nirikshak/packages/cli/node_modules/yargs/yargs.js:599:25)[39m
[90m    at /home/yash/Desktop/nirikshak/packages/cli/lib/index.js:83:14[39m
[90m    at Generator.next (<anonymous>)[39m
[90m    at fulfilled (/home/yash/Desktop/nirikshak/packages/cli/lib/index.js:6:58)[39m
"
`;
